rds-deployment-custom-image-rdsh/azuredeploy.json

```diff
@@ -1,161 +1,216 @@
+    },
+    "imageSKU": {
+      "type": "string",
+      "allowedValues": [
+        "2012-R2-Datacenter",
+        "Windows-Server-Technical-Preview"
+      ],
+      "metadata": {
+        "description": "Windows server SKU"
+      },
+      "defaultValue": "Windows-Server-Technical-Preview"
+    },
+    "rdshsourceImageVhdUri": {
+      "type": "string",
+      "metadata": {
+        "description": "The URI for source of the blob containing the custom image. For example https://johnrdshstore.blob.core.windows.net/johncontainer/MyWindowServerOS.vhd"
+      }
+    },
+    "rdshImageStorageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "This is the name of the your storage account where you stored the os custom image you want to use for rdsh machines. For example johnrdshstore"
+      }
+    },
+    "rdshImageResourceGroupName": {
+      "type": "string",
+      "metadata": {
+        "description": "This is the name of resource group where you stored the os custom image you want to use for rdsh machines. For example johnrdshStoreResourceGroup"
+      },
+      "defaultValue": "[resourceGroup().name]"
+    },
+    "numberOfRdshInstances": {
+      "type": "int",
+      "defaultValue": 1,
+      "metadata": {
+        "description": "Number of RemoteDesktopSessionHosts"
+      }
+    },
+    "rdshVmSize": {
+      "type": "string",
+      "allowedValues": [
+        "Standard_A0",
+        "Standard_A1",
+        "Standard_A2",
+        "Standard_A3",
+        "Standard_A4",
+        "Standard_A5",
+        "Standard_A6",
+        "Standard_A7",
+        "Standard_D1",
+        "Standard_D2",
+        "Standard_D3",
+        "Standard_D4",
+        "Standard_D5"
+      ],
+      "metadata": {
+          "description": "The size of the RDSH VMs"
+      },
+      "defaultValue": "Standard_A4"
+  }
+  },
     "variables": {
-        "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'rds', 'storage')]",
+        "gwdnsLabelPrefix": "[concat('gwd', resourceGroup().name)]",
+        "cbdnsLabelPrefix": "[concat('cbd', resourceGroup().name)]",
+        "storageAccountName": "iamuniquestore1",
+        "storageAccountType": "Standard_LRS",
+        "uniqueStorageAccountContainerName": "[concat('sc', resourceGroup().name)]",
         "imagePublisher": "MicrosoftWindowsServer",
         "imageOffer": "WindowsServer",
         "vnetAddressRange": "10.0.0.0/16",
         "subnetAddressRange": "10.0.0.0/24",
-        "dnsServerPrivateIp": "10.0.0.8",
-        "subnetName": "Subnet",
+        "dnsServerPrivateIp": "10.0.0.4",
         "subnet-id": "[concat(resourceId('Microsoft.Network/virtualNetworks',parameters('adVnetName')),'/subnets/',parameters('adSubnetName'))]",
         "publicIpRef": "publicIp",
+        "brokerIpRef": "brokerpublicIp",
+        "gatewayIpRef": "gatewaypublicIp",
         "assetLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/rds-deployment/",
```

Comment:
Please see the best practice around _artifactsLocation and _artifactsLocationSasToken: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md